cmake/generic.cmake

```diff
@@ -316,5 +316,5 @@ endfunction(go_test)
 # go_extern(target_name extern_source)
 # go_extern(go_redis github.com/hoisie/redis)
 function(go_extern TARGET_NAME)
-  add_custom_target(${TARGET_NAME} env GOPATH=${GOPATH} ${CMAKE_Go_COMPILER} get ${ARGN})
+  add_custom_target(${TARGET_NAME} env GOPATH=${GOPATH} ${CMAKE_Go_COMPILER} get -d ${ARGN})
```

Comment:
@typhoonzero I think when using `-u` the way we use symlink to link github.com/PaddlePaddle/Paddle in $GOPATH may break, since when using `-u` flag, it will try to update the local git repo. `-u` have the benefit of always having the latest code, but the build may break if dependency introduces breaking change. Maybe if we want to be explicit on the dependency version management, as you suggested, a vendor tool would be the best fit.
Maybe let's use without `-u` for now, and add a vendor tool?